### DISCUSSION TITLE
[A-L] (13 Febbraio 2024) Esercizio 6

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#410] [Esami A-L]
Consideriamo la seguente formula proposizionale in CNF:

### Fila A

$$
    \Large \\{\ \\{\neg p,\ \neg q\\},\ \\{\neg p,\ q\\},\ \\{p,\ \neg q\\},\ \\{p,\ q\\},\ \\{s,\ p\\},\ \\{\neg s,\ p,\ \neg q\\}\ \\}.
$$

### Fila B

$$
    \Large \\{\ \\{p,\ q,\ r\\},\ \\{p,\ q,\ \neg r\\},\ \\{p,\ \neg q,\ r\\},\ \\{p,\ \neg q,\ \neg r\\},\ 
$$

$$
     \Large \\{\neg p,\ q,\ r\\},\ \\{\neg p,\ q,\ \neg r\\},\ \\{\neg p,\ \neg q,\ r\\},\ \\{\neg p,\ \neg q,\ \neg r\\}\ \\}.
$$

### Entrambe le file

1. Applicare tre volte la regola di Risoluzione alla formula. [2.5 punti]

2. La formula è soddisfacibile? (Se si risponde "SI" definire un assegnamento che la soddisfa, se si risponde "NO" dimostrare l’insoddisfacibilità usando la regola di Risoluzione.) [2.5 punti]

**Comment by DavideGalilei** [comment]
### 🅰️ Fila A

Termini:
1. $\Large{\\{ \lnot{p}, \lnot{q} \\}}$
2. $\Large{\\{\lnot{p}, q\\}}$
3. $\Large{\\{p, \lnot{q}\\}}$
4. $\Large{\\{p, q\\}}$
5. $\Large{\\{s, p\\}}$
6. $\Large{\\{\lnot{s}, p, \lnot{q}\\}}$

Scelgo i termini 1 e 2, ed elimino la q
7. $\Large{\\{ \lnot{p} \\}}$

Scelgo i termini 3 e 4, ed elimino la q:
8. $\Large{\\{p\\}}$

Scelgo i termini 7 ed 8, ma sono uno in contrasto con l'altro, quindi la soluzione è $UNSAT$ $\square$

### 🅱️ Fila B

Termini:
1. $\Large{\\{p, q, r\\}}$
2. $\Large{\\{p, q, \lnot{r}\\}}$
3. $\Large{\\{p, \lnot{q}, r\\}}$
4. $\Large{\\{p, \lnot{q}, \lnot{r}\\}}$
5. $\Large{\\{\lnot{p}, q, r\\}}$
6. $\Large{\\{\lnot{p}, q, \lnot{r}\\}}$
7. $\Large{\\{\lnot{p}, \lnot{q}, r\\}}$
8. $\Large{\\{\lnot{p}, \lnot{q}, \lnot{r}\\}}$

Scelgo i termini 1 e 2, ed elimino la r:
9. $\Large{\\{p, q\\}}$

Scelgo i termini 3 e 4, ed elimino la r:
10. $\Large{\\{p, \lnot{q}\\}}$

Scelgo i termini 5 e 6, ed elimino la r:
11. $\Large{\\{\lnot{p}, q\\}}$

Scelgo i termini 7 e 8, ed elimino la r:
12. $\Large{\\{\lnot{p}, \lnot{q}\\}}$

Scelgo i termini 9 e 10, ed elimino la q:
13. $\Large{\\{p\\}}$

Scelgo i termini 11 e 12, ed elimino la q:
14. $\Large{\\{\lnot{p}\\}}$

Scelgo i termini 13 e 14, e noto che sono in contrasto tra di loro, quindi la soluzione è $UNSAT$ $\square$
